### ISSUE TITLE
Async task fails on 2.1 with WinRMTransportError even when ignore_errors is set to true

### ISSUE THREAD

**Issue by rjosephwright** [ansible/ansible#16077]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = 
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

No ansible.cfg used. ANSIBLE_ROLES_PATH and ANSIBLE_HOST_KEY_CHECKING environment variables are set.
##### OS / ENVIRONMENT

Running from Ubuntu 15.10
Managing Windows 2008r2 and 2012r2
##### SUMMARY

I have a `win_package` task which reboots the machine when it finishes installing a package. The task uses async with `ignore_errors` set to `true`, then waits for the machine to come back on the next task. On previous Ansible versions, this worked fine, but after upgrading to 2.1, the task fails and the playbook does not continue.
##### STEPS TO REPRODUCE

This is the failing task and the one that follows:

```
- name: Install Trend Micro SecureCloud
  win_package:
    path="C:\SecureCloudAgentSilentSetup_Release_Win32_en_3.7.0_1107.exe"
    product_id="{2301FEB0-7D71-4DB4-A164-A95A7DDBD3B3}"
    arguments="/q"
  async: 0
  poll: 0
  ignore_errors: true
  when: sc_installed | failed

- name: Waiting for server to come back
  local_action: wait_for host={{ inventory_hostname }} state=started delay=30 timeout=300
  when: sc_installed | failed
```
##### EXPECTED RESULTS

I expected any errors that occur during the task `Install Trend Micro SecureCloud` to be ignored, and the task `Waiting for server to come back` to wait. This is the behavior with previous versions of Ansible that I used.
##### ACTUAL RESULTS

Task fails but error is not ignored.

```
TASK [base-windows : Install Trend Micro SecureCloud] **************************
task path: /home/joswright/Workspace/base-windows/tasks/main.yml:221
<10.x.x.x> ESTABLISH WINRM CONNECTION FOR USER: Administrator on PORT 5985 TO 10.x.x.x
<10.x.x.x> EXEC Set-StrictMode -Version Latest
(New-Item -Type Directory -Path $env:temp -Name "ansible-tmp-1464737336.12-201371408724451").FullName | Write-Host -Separator '';
<10.x.x.x> PUT "/tmp/tmpjy_Xb5" TO "C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1464737336.12-201371408724451\win_package.ps1"
<10.x.x.x> EXEC Set-StrictMode -Version Latest
Try
{
& 'C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1464737336.12-201371408724451\win_package.ps1'
}
Catch
{
$_obj = @{ failed = $true }
If ($_.Exception.GetType)
{
$_obj.Add('msg', $_.Exception.Message)
}
Else
{
$_obj.Add('msg', $_.ToString())
}
If ($_.InvocationInfo.PositionMessage)
{
$_obj.Add('exception', $_.InvocationInfo.PositionMessage)
}
ElseIf ($_.ScriptStackTrace)
{
$_obj.Add('exception', $_.ScriptStackTrace)
}
Try
{
$_obj.Add('error_record', ($_ | ConvertTo-Json | ConvertFrom-Json))
}
Catch
{
}
Echo $_obj | ConvertTo-Json -Compress -Depth 99
Exit 1
}
Finally { Remove-Item "C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1464737336.12-201371408724451" -Force -Recurse -ErrorAction SilentlyContinue }
Traceback (most recent call last):
  File "/home/joswright/.pyenv/versions/2.7.11/envs/bossimage/lib/python2.7/site-packages/ansible/plugins/connection/winrm.py", line 271, in exec_command
    result = self._winrm_exec(cmd_parts[0], cmd_parts[1:], from_exec=True)
  File "/home/joswright/.pyenv/versions/2.7.11/envs/bossimage/lib/python2.7/site-packages/ansible/plugins/connection/winrm.py", line 234, in _winrm_exec
    self.protocol.cleanup_command(self.shell_id, command_id)
  File "/home/joswright/.pyenv/versions/2.7.11/envs/bossimage/lib/python2.7/site-packages/winrm/protocol.py", line 290, in cleanup_command
    rs = self.send_message(xmltodict.unparse(rq))
  File "/home/joswright/.pyenv/versions/2.7.11/envs/bossimage/lib/python2.7/site-packages/winrm/protocol.py", line 193, in send_message
    return self.transport.send_message(message)
  File "/home/joswright/.pyenv/versions/2.7.11/envs/bossimage/lib/python2.7/site-packages/winrm/transport.py", line 138, in send_message
    raise WinRMTransportError('http', ex.reason)
WinRMTransportError: 500 WinRMTransport. [Errno 111] Connection refused
fatal: [10.x.x.x]: UNREACHABLE! => {"changed": false, "msg": "failed to exec cmd PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -EncodedCommand UwBlAHQALQBTAHQAcgBp<truncated>HQAaQBuAHUAZQAgAH0A", "unreachable": true}
```

**Comment by nitzmahone** [comment]
Async is not currently supported on Windows at all yet- roadmapped for 2.2 (and currently being prototyped by yours truly). 

That said, async isn't really the right way to do this anyway- I'd suggest suppressing the reboot from win_package, then conditionally calling [win_reboot](http://docs.ansible.com/ansible/win_reboot_module.html) instead if it registered a change (or supply your own "needs reboot" criteria as necessary). :)

**Comment by rjosephwright** [comment]
This used to work before 2.1, so there has been a regression in functionality. To my knowledge, there is no way to suppress a reboot in my case, so I will probably have to keep using 2.0 until newer versions start working again. Also, the approach I am using is widely documented, even on [ansible.com](https://support.ansible.com/hc/en-us/articles/201958037-Reboot-a-server-and-wait-for-it-to-come-back), so if it is the wrong way, some documentation needs to get updated :).

**Comment by nitzmahone** [comment]
Can you point me at the docs in question? Async requires major changes to the entire Ansible Windows exec process to allow processes to survive the shutdown of a WinRM job, so I highly doubt it's ever worked _there_. Use of async is documented for *nix stuff, but shouldn't be for Windows.

**Comment by nitzmahone** [comment]
PS, I looked: the Trend Micro agent in question is using an MSI with ScheduleReboot under the covers, so there's probably a way to pass the standard MSI REBOOT=ReallySuppress through their little exe wrapper to make it behave nicely (maybe ask them). Worst case, you can run the wrapper with /C and /T to extract locally then use win_package to install the MSI and pass REBOOT=ReallySuppress that way. Regardless, there are numerous ways you should be able to manage the reboot yourself, which is the only safe way. Even once async is working for Windows, there's still a race there between the async registration and the reboot- if it happens too fast, Ansible will still be upset. :)

**Comment by rjosephwright** [comment]
I gave a link in the comment above, although it is not specific to Windows. The [async documentation](http://docs.ansible.com/ansible/playbooks_async.html) doesn't mention that Windows is unsupported, so it would be helpful if that were mentioned.

I guess when I used async, it wasn't actually "working", but I did get the result I wanted, which was to wait for the machine to reboot. I am no longer able to use `ignore_errors: true`, because Ansible just ignores it and crashes now. Thanks for the advice on alternative approaches, I'll look into reworking it.
